src/tsolvers/lasolver/LASolver.cc

```diff
@@ -332,6 +352,78 @@ void LASolver::informNewSplit(PTRef tr)
     updateBound(tr);
 }
 
+TRes LASolver::checkIntegersAndSplit() {
+
+    bool nonint_models = false;  // Keep track whether non-integer models are seen
+
+    for (LVRef x : int_vars) {
+
+        if (!isModelInteger(x)) {
+            nonint_models = true;
+            //Prepare the variable to store a splitting value
+            opensmt::Real c;
+
+            // if val of int variable is not int, set it to integer by getting floor (c) and ceiling (c+1)
+            // if real part of int var is int, then delta must be non-zero
+            Delta x_val = simplex.getValuation(x);
+            if (!x_val.R().isInteger()) {
+                c = x_val.R().floor();
+            } else { //but if the value from LRA solver returned is integer(which is here subset of real), then we consider delta part
+                assert(x_val.D() != 0);
+                if (x_val.D() < 0) {
+                    c = x_val.R() - 1;
+                } else {
+                    c = x_val.R();
+                }
+            }
+
+            // We might have this blocked already, and then the solver should essentially return "I don't know, please go ahead".
+            if (cuts[getVarId(x)].find(c) != cuts[getVarId(x)].end()) {
+                continue;
+            }
+            cuts[getVarId(x)][c] = true;
```

Comment:
Do we need these `cuts`. They seem a bit weird. But that probably need more investigation not related to this PR.

Reply:
OK, let's look into that at a later time.  And not forget it.